### PR TITLE
Add support for image creation and image listing

### PR DESCRIPTION
**Pull request by lukedirtwalker**
Image create only supports the pull usecase.
Also fix a typo in container/list.go.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Thanks for working on this.

I think it would be nice to have separate functions for pull vs load or whatever else we might have.

File: image/create.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	resp, err := httputil.DoRequest(ctx, func(ctx context.Context) (*http.Response, error) {
+		return s.tr.Do(ctx, http.MethodPost, version.Join(ctx, "/images/create"), withCreateConfig)
+	})
+	if err != nil {
+		return err
+	}
+	// Reading all is required otherwise the docker daemon considers the context
+	// as cancelled and the pulling is aborted.
+	_, err = ioutil.ReadAll(resp.Body)
```
Comment (cpuguy83) [inline image/create.go:50]: Docker is streaming a response back with all the pull data.
That's why it needs to be consumed before closing.

We probably want to return the stream as a `<-chan <type>` so callers can consume that.

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

**Comment by lukedirtwalker** [comment]
> Thanks for working on this.

Thanks for the review

> I think it would be nice to have separate functions for pull vs load or whatever else we might have.

Yeah that sounds nice, do you know where I find out which parameters are relevant for each use case.
So far I used https://docs.docker.com/engine/api/v1.41/#operation/ImageCreate but from that it isn't clear.

File: image/create.go (hunk 1 above)
Comment (lukedirtwalker) [reply image/create.go:50]: Are the responses documented somewhere? https://docs.docker.com/engine/api/v1.41/#operation/ImageCreate doesn't say anything about it :/

**Comment by cpuguy83** [comment]
https://github.com/moby/moby/blob/f95be5e2f39f2f35501667fbf88210627ac3fe37/api/server/router/image/image_routes.go#L56-L74

**Comment by cpuguy83** [comment]
FYI, I added image pull support here: https://github.com/cpuguy83/go-docker/commit/55a58698b2e7740cfbc4e24ead9078ed18c4faf8

**Comment by lukedirtwalker** [comment]
Closing this, as this was just a PR from our main branch. I'll create follow up PRs for functionality that is missing.

**Pull request closed without merging**